services/s3/src/main/resources/codegen-resources/customization.config

```diff
@@ -172,5 +172,14 @@
       "clientConfiguration"
     ]
   },
+  "additionalBuilderMethods": [
```

Comment:
Can we specify this is for async only, something like `additionalBuilderMethodsForAsyncClient`?

Reply:
Why is it restricted to async?

Reply:
Because S3 CRT client is an async client and we don't support it for sync code path for now

Reply:
This use case, sure, but the additionalBuilderMethods functionality looks general enough to be used for a sync use case too.

Reply:
We only need it for async client though for now. I guess we can add a flag in the `additionalBuilderMethods` to indicate if it's for sync or async. I'm fine with either way since it's an internal API.

Reply:
Added a new field to indicate the clientType supported